app/controllers/reviews_controller.rb

```diff
@@ -21,6 +26,13 @@ def new
 
   def create
     @review = parent.reviews.new(review_params)
+
+    if current_user.nil?
```

Comment:
you don't want to be checking if a user is present in your controllers, extract that logic into models and views